File: frontend/src/api/advisors.ts
import { apiFetch } from './client'

export interface AdvisorItem {
  id: string
  owner_id?: string
  data: Record<string, any>
  created_at: string
  updated_at?: string
}

export interface AdvisorCreate { data: Record<string, any>; owner_id?: string }
export interface AdvisorUpdate { data?: Record<string, any> }
export interface AdvisorsImport { items: Array<Record<string, any>>; owner_id?: string }

export async function getAdvisors(): Promise<AdvisorItem[]> {
  return apiFetch<AdvisorItem[]>(`/advisors`)
}

export async function createAdvisorItem(body: AdvisorCreate): Promise<AdvisorItem> {
  return apiFetch<AdvisorItem>(`/advisors`, { method: 'POST', body: JSON.stringify(body) })
}

export async function updateAdvisorItem(id: string, body: AdvisorUpdate): Promise<AdvisorItem> {
  return apiFetch<AdvisorItem>(`/advisors/${id}`, { method: 'PUT', body: JSON.stringify(body) })
}

export async function deleteAdvisorItem(id: string): Promise<null> {
  return apiFetch<null>(`/advisors/${id}`, { method: 'DELETE' })
}

export async function clearAdvisors(): Promise<null> {
  return apiFetch<null>(`/advisors/clear`, { method: 'DELETE' })
}

export async function importAdvisors(body: AdvisorsImport): Promise<AdvisorItem[]> {
  return apiFetch<AdvisorItem[]>(`/advisors/import`, { method: 'POST', body: JSON.stringify(body) })
}

File: backend/app/schemas/advisors.py
from pydantic import BaseModel
from typing import Optional, Any
from datetime import datetime


class AdvisorBase(BaseModel):
    data: dict[str, Any]


class AdvisorCreate(AdvisorBase):
    owner_id: Optional[str] = None


class AdvisorUpdate(BaseModel):
    data: Optional[dict[str, Any]] = None


class AdvisorItem(BaseModel):
    id: str
    owner_id: Optional[str] = None
    data: dict[str, Any]
    created_at: datetime
    updated_at: Optional[datetime] = None

    class Config:
        from_attributes = True


class AdvisorsImport(BaseModel):
    items: list[dict]
    owner_id: Optional[str] = None








